includes/formatting.php

```diff
@@ -247,3 +269,62 @@ function set_attachment_author( $post ) {
 
 	return $post;
 }
+
+/**
+ * When importing manipulated images, also side-load the original version of the media.
+ *
+ * Airstory currently uses Cloudinary to host images and allow users to manipulate (scale, rotate,
+ * etc.) using a series of modifiers in the URL path.
+ *
+ * An example URL with modifiers would be:
+ *
+ *   https://cloudinary.com/airstory/image/upload/c_scale,w_0.1/v1/prod/iXXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/image.jpg
+ *
+ * The "/c_scale,w_0.1/" portion of the path tells Cloudinary to scale the image to 10% of the
+ * original width.
+ *
+ * The equivalent transformed image with the images.airstory.co domain would be:
+ *
+ *   https://images.airstory.co/c_scale,w_0.1/v1/prod/iXXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/image.jpg
+ *
+ * @see http://cloudinary.com/documentation/image_transformations for a full list of available
+ *      Cloudinary modifier arguments.
+ *
+ * While we want to respect the version originally exported from Airstory, there's value in
+ * also capturing the original media, in case the user wants to re-generate thumbnails later.
+ *
+ * @param string $url The URL for the image, as hosted on Cloudinary.com.
+ * @param int    $post_id  The post the newly-uploaded image should be attached to.
+ * @param array  $metadata Additional post meta keys to assign once the attachment post bas been
+ *                         created. These keys and values are assumed to be sanitized.
+ */
+function retrieve_original_media( $url, $post_id, $metadata ) {
+	$url_components = array_merge( array( 'host' => '', 'path' => '' ), (array) parse_url( $url ) );
+	$url_components = array_map( 'strtolower', $url_components );
+
+	// Only operate on Cloudinary-hosted images.
+	if ( ! in_array( $url_components['host'], array( 'res.cloudinary.com', 'images.airstory.co' ), true ) ) {
```

Comment:
This definitely does duplicate the list of domains (thanks, PHP < 7 not letting us use arrays in constants!), but for a slightly different purpose: before switching to images.airstory.co, the res.cloudinary.com URLs were being used for resized/cropped content, and these URLs still exist in older Airstory content. This function, born out of #27, is only meant to supplement the media imported by the plugin from the two canonical media hosting domains for Airstory.

If an Airstory user was including media from another "use URL parameters to manipulate the image" service, they'd likely be better off adding a new callback to the `airstory_sideload_single_image` action than hacking apart this function (which returns early if the logic contained is not applicable).